sqlglot/executor/__init__.py

```diff
@@ -23,9 +24,11 @@ def execute(sql, schema, read=None):
                 3. {catalog: {db: {table: {col: type}}}}
         read (str): the SQL dialect to apply during parsing
             (eg. "spark", "hive", "presto", "mysql").
+        tables (dict[str, list[dict]|sqlglot.executor.Table]): additional tables to register
     Returns:
         sqlglot.executor.Table: Simple columnar data structure.
     """
+    tables = _ensure_tables(tables)
```

Comment:
given tables, you should be able to infer the schema

Reply:
Unless the tables are empty